planner/core/plan_cache.go

```diff
@@ -285,6 +285,7 @@ func getCachedPlan(sctx sessionctx.Context, isNonPrepared bool, cacheKey kvcache
 		core_metrics.GetPlanCacheHitCounter(isNonPrepared).Inc()
 	}
 	stmtCtx.SetPlanDigest(stmt.NormalizedPlan, stmt.PlanDigest)
+	stmtCtx.StmtHints = cachedVal.sessVarHints
```

Comment:
The `PointGet` short path function `getCachedPointPlan` should be updated as well.

Reply:
Fixed.